metricbeat/module/mongodb/replstatus/info.go

```diff
@@ -92,28 +86,36 @@ func getReplicationInfo(client *mongo.Client) (*oplogInfo, error) {
 	}, nil
 }
 
-func getOpTimestamp(collection *mongo.Collection, sort string) (uint32, error) {
-	opt := options.Find().SetSort(bson.D{{Key: sort, Value: 1}})
-	cursor, err := collection.Find(context.Background(), bson.D{}, opt)
-	if err != nil {
-		return 0, fmt.Errorf("could not get cursor on collection '%s': %w", collection.Name(), err)
+func getOpTimestamp(collection *mongo.Collection) (uint32, uint32, error) {
+
+	// Find both first and last timestamps using $min and $max
+	pipeline := bson.A{
+		bson.M{"$group": bson.M{"_id": 1, "minTS": bson.M{"$min": "$ts"}, "maxTS": bson.M{"$max": "$ts"}}},
 	}
 
-	if !cursor.Next(context.Background()) {
-		return 0, errors.New("objects not found in local.oplog.rs")
+	cursor, err := collection.Aggregate(context.Background(), pipeline)
+	if err != nil {
+		return 0, 0, fmt.Errorf("could not get operation timestamps in op log: %w", err)
 	}
+	defer cursor.Close(context.Background())
 
-	var opTime map[string]interface{}
-	if err = cursor.Decode(&opTime); err != nil {
-		return 0, fmt.Errorf("error decoding response: %w", err)
+	var result struct {
+		MinTS time.Time `bson:"minTS"`
+		MaxTS time.Time `bson:"maxTS"`
 	}
 
-	ts, ok := opTime["ts"].(primitive.Timestamp)
-	if !ok {
-		return 0, errors.New("an expected timestamp was not found")
+	if cursor.Next(context.Background()) {
+		if err := cursor.Decode(&result); err != nil {
+			return 0, 0, fmt.Errorf("error decoding response for timestamps: %w", err)
+		}
+	} else {
+		return 0, 0, errors.New("no documents found in op log")
```

Comment:
A more readable version of this could be:
```suggestion
	if !cursor.Next(context.Background()) {
	        return 0, 0, errors.New("no documents found in op log")
	} 
	if err := cursor.Decode(&result); err != nil {
	       return 0, 0, fmt.Errorf("error decoding response for timestamps: %w", err)
	}
```